### PR TITLE
Transfer APIM 3.0.0 changes from lakwarus fork

### PR DESCRIPTION
**Pull request by chamilad**
This PR transfers the APIM 3.0.0. K8S and Dockerfile artifacts from @lakwarus's fork https://github.com/lakwarus/kubernetes-apim to branch `3.0.0`. After this PR is merged, it will be convenient to do changes to the above artifacts as the milestone releases are done.

### CODE REVIEW

File: pattern-1/activemq-deployment.yaml [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+  template:
+    metadata:
+      labels:
+        app: activemq
+        tier: prod
+    spec:
+      containers:
+      - image: docker.wso2.com/activemq 
+        name: activemq
+        imagePullPolicy: Never
```
Comment (pubudu538) [inline pattern-1/activemq-deployment.yaml]: We can change this imagePullPolicy to ifNotPresent. Users may configure docker registry and will not always use docker.wso2.com. We can use this policy in all the deployments.

File: pattern-1/deploy.sh [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+oc create -f activemq-service.yaml
+oc create -f activemq-deployment.yaml
+sleep 5
+oc create -f key-manager-service.yaml
+oc create -f key-manager-deployment.yaml
+sleep 10
+oc create -f api-core-service.yaml
+sleep 5
+oc create -f api-core-deployment.yaml
+oc create -f api-gateway-deployment.yaml
```
Comment (pubudu538) [inline pattern-1/deploy.sh]: Shall we deploy services first?  Then we can avoid sleep time here.. We only need to wait until MB gets started. KM, Core and Gateway can parallelly start.

File: pattern-1/undeploy.sh [hunk 3]
```diff
@@ -0,0 +1,9 @@
+#!/bin/bash
+
+oc delete -f activemq-service.yaml
+oc delete -f activemq-deployment.yaml
+oc delete -f key-manager-service.yaml
+oc delete -f key-manager-deployment.yaml
+oc delete -f api-core-service.yaml
+oc delete -f api-core-deployment.yaml
+oc delete -f api-gateway-deployment.yaml
```
Comment (pubudu538) [inline pattern-1/undeploy.sh]: Shall we rename deploy and undeploy scripts as deploy-openshift,etc ? Also, shall we add a label called pattern=wso2apim-pattern-1 in every resource? Then we can remove resources in one command as in https://github.com/wso2/kubernetes-apim/blob/2.1.0/pattern-1/artifacts/undeploy-openshift.sh

**Comment by chamilad** [comment]
@pubudu538 Thanks for the input! I will do the changes in the subsequent commits. Could this PR be merged to branch `3.0.0` with Lakmal's commit, so that next PRs can be opened?

**Comment by pubudu538** [comment]
I will merge the PR based on the above comment.

**Pull request merged**